src/libYARP_OS/src/ThreadImpl.cpp

```diff
@@ -359,6 +359,7 @@ bool ThreadImpl::isClosing()
 
 bool ThreadImpl::isRunning()
 {
+    assert(threadMutex);
```

Comment:
This assert does not make sense, it should be everywhere...
This mutex is protecting the access to a bool and to an int, instead these varables should be made atomic.

Reply:
i agree with you with making atomic the variables protected by this mutex.. however let me explain the reason for this assert:

the pointer is not checked before accessing it. there some situation where it is a nullptr and accessing it cause a crash. at first i've simply checked the pointer and returned false but @barbalberto rightly said that this kind of situation should not happen at all (a yarp thread is used without the "yarp::os::Network" singleton in memory) and the program should not continue... so for this specific case i thought an assert was better than a segfault.